app/models/compute_resources/foreman/model/gce.rb

```diff
@@ -39,10 +39,8 @@ def email=(email)
       attrs[:email] = email
     end
 
-    #TODO: allow to select public / private ip address that foreman tries to find
-    def provided_attributes
-      super.merge({ :ip => :public_ip_address })
```

Comment:
Are you sure you should remove this?  When it's removed, Foreman makes the *user* supply the IP address, which won't be appropriate as it has to come from one of their configured subnets.  With it here, Foreman pulls the IP address from the GCE VM.